oncotator/utils/FieldMapCreator.py

```diff
@@ -0,0 +1,111 @@
+        hdr = header.lower()
+        alternatives = alternative_dict.get(hdr, [])
+        full_possibilities = [header] + alternatives
+        result = None
+
+        if not deprioritize_input_annotations:
+            for p in full_possibilities:
+                if p in annotation_names:
+                    result = p
+                    break
+        else:
+            second_choice = None
+            is_choosing_second_choice = True
+            for p in full_possibilities:
+                if p in annotation_names and m.getAnnotation(p).getDatasource() != "INPUT":
+                    result = p
+                    is_choosing_second_choice = False
+                    break
+                if second_choice is None and p in annotation_names:
+                    second_choice = p
+
+            if is_choosing_second_choice:
+                result = second_choice
+
+        return result
+
+
+    @staticmethod
+    def create_field_map(headers, m, alternative_dict, is_render_internal_fields=True,
+                            exposed_fields=None, prepend="i_", deprioritize_input_annotations=False):
+        """
+        Create a mapping for output header to the best input annotation.
+
+        This can handle prepend fields (attach the prepend to internal fields), exposed fields (ones not in the list of headers, but should not have a prepend),
+
+        :param is_render_internal_fields: Whether annotations not assigned to headers (or superseded by other annotations) should be included in this map.
+        :type is_render_internal_fields bool
+        :param exposed_fields: list of fields that, if found, should never receive a prepend.
+        :param prepend: The prepend to put on internal fields, if any are detected.  If is_render_internal_fields is False, this parameter does nothing.
+        :param deprioritize_input_annotations: If an annotation with the exact name of the header is found AND it has a datasource of "INPUT",
+            use one the annotations instead.  This is useful in cases where we want to reannotate.  This effectively handles aliases.
+        :param headers:  List of headers that need to be populated for rendering.  For example, the columns in a TCGA MAF
+        :param m: MutationData to scrape available annotations
+        :param alternative_dict: Dictionary of header to list of annotations.  Usually, created from a config file.
+        :return: dict of header:annotation name (one annotation name) that should be used for this output rendering.
+        """
+        result = dict()
+        if prepend is None:
+            prepend = ""
+
+        if exposed_fields is None:
+            exposed_fields = set()
+
+        # Process each header and find the first alias.  If an annotation exists with the exact same name as the header
+        #   use that unless deprioritization is in effect.
+        annotation_names = sorted(list(set(m.keys() + m.getAttributeNames())))
```

Comment:
Should this be extracted as a method on mutation?  This is the second time this line appears in this class.